app/models/collection.rb

```diff
@@ -14,4 +14,23 @@ class Collection < ApplicationRecord
   def review_enabled?
     reviewers.present?
   end
+
+  state_machine initial: :first_draft do
+    after_transition on: :begin_deposit do |collection, _transition|
```

Comment:
Heads-up to @sul-dlss/infrastructure-team that we're now leaning on our state machine library to handle side effects we want state changes to trigger. Note that these are automatically done in a transaction